### PR TITLE
feat: [CDS-58027]: Added support for Env group at Env & Artifact card and Env Table view

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
#### Screenshots

https://user-images.githubusercontent.com/98325173/233254151-7d8fe637-1f65-43cd-b559-3c8574e12b58.mov

Screenshot 2023-04-20 at 9 09 54 AM

Screenshot 2023-04-20 at 9 10 03 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailDriftTable.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  isArtifactView
+}: {
+  data: ArtifactDeploymentDetail[]
+  isArtifactView: boolean
+  artifactName?: string
+}): JSX.Element {
+  const { getString } = useStrings()
+  const artifactDetail = defaultTo(data, [])
+
+  const RenderEnvName: Renderer<CellProps<ArtifactDeploymentDetail>> = ({ row }) => {
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailDriftTable.tsx:41]: This should be defined outside

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailDriftTable.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+          e.stopPropagation()
+          handleClick()
+        }}
+      >
+        {envName || '--'}
+      </Text>
+    )
+  }
+
+  const RenderLatestVersion: Renderer<CellProps<ArtifactDeploymentDetail>> = ({ row }) => {
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailDriftTable.tsx:73]: define all cells outside

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsSummaryV2.module.scss [hunk 3]
```diff
@@ -7,21 +7,70 @@
...
     display: flex;
     flex-direction: row;
     align-items: center;
-
-    > p {
-      width: 240px;
+    justify-content: space-between;
+
+    .statusIcon {
+      background: rgba(11, 11, 13, 0.1);
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsSummaryV2.module.scss:24]: Do we have a color var for this?

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvView.tsx [hunk 4]
```diff
@@ -68,12 +84,43 @@ function createGroups(
...
+  >
   env?: EnvCardProps
-  selectedEnv?: string
+  selectedEnv?: {
+    envId?: string
+    isEnvGroup: boolean
+  }
+}
+
+const latestTimeAndArtifact = (data: ArtifactDeploymentDetail[] | undefined): ArtifactDeploymentDetail => {
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvView.tsx:111]:
```suggestion
const getLatestTimeAndArtifact = (data: ArtifactDeploymentDetail[] | undefined): ArtifactDeploymentDetail => {
```

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvView.tsx [hunk 5]
```diff
@@ -391,12 +544,63 @@ export function ServiceDetailsEnvView(props: ServiceDetailsEnvViewProps): React.
           )
         })
 
+  const dropDownOptions = React.useMemo(() => {
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvView.tsx:547]:
```suggestion
  const dropdownOptions = React.useMemo(() => {
```

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvView.tsx [hunk 6]
```diff
@@ -391,12 +544,63 @@ export function ServiceDetailsEnvView(props: ServiceDetailsEnvViewProps): React.
...
+
+      envRefetch({
+        queryParams,
+        body: {
+          filterType: 'Environment',
+          environmentTypes: sortFilter
+        }
+      })
+    },
+    [envRefetch]
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvView.tsx:587]: FYI, queryParams are not added to the dependencies

**Review by rameshx — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailDriftTable.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        isArtifactView
+      },
+      {
+        id: 'lastDeployed',
+        width: '30%',
+        Cell: RenderLastDeployedTime,
+        Header: getString('cd.serviceDashboard.lastDeployment').toUpperCase()
+      }
+    ],
+    [getString]
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailDriftTable.tsx:156]: missing dependencies

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsSummaryV2.module.scss [hunk 8]
```diff
@@ -414,3 +470,92 @@
...
+  margin-top: 4px;
+  margin-bottom: 6px;
+  border-top: 1px solid var(--grey-200);
+}
+
+.envGroupTooltip {
+  padding: var(--spacing-5) !important;
+  width: 240px;
+
+  div[id='envGrpText'] {
```
Comment (rameshx) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsSummaryV2.module.scss:557]: nit
```suggestion
  #envGrpText {
```

**Review by rameshx — APPROVED** [review]

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsSummaryV2.module.scss (hunk 8 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsSummaryV2.module.scss:557]: Thanks will fix in the next pr.

**Review by Eric761 — APPROVED** [review]

**Pull request merged**